flatpak compiler warnings

```
[159/187] Building CXX object plugins/flatpak/CMakeFiles/plugin_flatpak.dir/flatpak_shim.cc.o
/mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.cc:1264:24: warning: variable 'installed' set but not used [-Wunused-but-set-variable]
 1264 |   FlatpakInstalledRef* installed = nullptr;
      |                        ^
/mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.cc:1910:14: warning: lambda capture 'this' is not used [-Wunused-lambda-capture]
 1910 |             [this, completion_callback, installation, found_app_name,
      |              ^~~~~
/mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.cc:3731:54: warning: unused parameter 'transaction' [-Wunused-parameter]
 3731 | void FlatpakShim::OnNewOperation(FlatpakTransaction* transaction,
      |                                                      ^
/mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.cc:3793:59: warning: unused parameter 'transaction' [-Wunused-parameter]
 3793 | void FlatpakShim::OnOperationComplete(FlatpakTransaction* transaction,
      |                                                           ^
/mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.cc:3839:25: warning: unused parameter 'transaction' [-Wunused-parameter]
 3839 |     FlatpakTransaction* transaction,
      |                         ^
In file included from /mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.cc:17:
/mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.h:427:18: warning: private field 'plugin_' is not used [-Wunused-private-field]
  427 |   FlatpakPlugin* plugin_;
      |                  ^
/mnt/raid10/workspace-automation/app/ivi-homescreen-plugins/plugins/flatpak/flatpak_shim.h:428:29: warning: private field 'messenger_' is not used [-Wunused-private-field]
  428 |   flutter::BinaryMessenger* messenger_;
      |                             ^
```

--- a/plugins/flatpak/README.md
+++ b/plugins/flatpak/README.md
@@ -2,16 +2,9 @@
 
 ### Status
 
-#### Validated Dart API
-
-```
-getVersion
-getDefaultArch
-getSupportedArches
-getSystemInstallations
-getUserInstallation
-getApplicationsInstalled
-```
+* Applications can be installed
+* Progress is display during installation
+* Applications can be started in sandbox
 
 ### Ubuntu Package Dependency
 
@@ -38,12 +31,12 @@ flatpak run org.gnome.Todo
 
 ### Flutter code
 
-https://github.com/meta-flutter/flatpak_flutter
+https://github.com/toyota-connected/tcna-packages/tree/v2.0/packages/flatpak
 
 ### Generate message.g.h and messages.g.cc
 
-    git clone https://github.com/meta-flutter/flatpak_flutter
-    cd flatpak_flutter
+    git clone https://github.com/toyota-connected/tcna-packages/tree/v2.0/packages/flatpak
+    cd packages/flatpak
     dart run pigeon --input pigeons/messages.dart
 
 ### Flatpak API reference

--- a/plugins/flatpak/flatpak_shim.cc
+++ b/plugins/flatpak/flatpak_shim.cc
@@ -1261,8 +1261,6 @@ void FlatpakShim::ApplicationUninstall(
 
   spdlog::debug("[FlatpakPlugin] Uninstalling application: {}", id);
 
-  FlatpakInstalledRef* installed = nullptr;
-  std::string full_ref;
   GError* error = nullptr;
   auto installation = flatpak_installation_new_user(nullptr, &error);
   if (error) {
@@ -1316,9 +1314,7 @@ void FlatpakShim::ApplicationUninstall(
         found_arch = ref_arch ? ref_arch : "";
         found_branch = ref_branch ? ref_branch : "";
         app_found = true;
-        full_ref = full_app_id;
         g_object_ref(ref);
-        installed = ref;
         break;
       }
     }
@@ -1359,9 +1355,7 @@ void FlatpakShim::ApplicationUninstall(
             found_arch = ref_arch ? ref_arch : "";
             found_branch = ref_branch ? ref_branch : "";
             app_found = true;
-            full_ref = full_app_id;
             g_object_ref(ref);
-            installed = ref;
             break;
           }
         }
@@ -1884,8 +1878,8 @@ void FlatpakShim::ApplicationStart(
 
   check_runtime(
       installed, installation, strand,
-      [this, completion_callback, installed, installation, found_app_name,
-       found_arch, found_branch, portal_manager,
+      [completion_callback, installed, installation, found_app_name, found_arch,
+       found_branch, portal_manager,
        strand_ptr = &strand](const ErrorOr<bool>& runtime_result) {
         if (runtime_result.has_error()) {
           spdlog::error("[FlatpakPlugin] Runtime check failed: {}",
@@ -1907,8 +1901,8 @@ void FlatpakShim::ApplicationStart(
 
         create_sandbox(
             installed, *strand_ptr,
-            [this, completion_callback, installation, found_app_name,
-             found_arch, found_branch, portal_manager, installed,
+            [completion_callback, installation, found_app_name, found_arch,
+             found_branch, portal_manager, installed,
              strand_ptr](bool sandbox_success) {
               if (!sandbox_success) {
                 spdlog::error("[FlatpakPlugin] Failed to create sandbox");
@@ -3728,7 +3722,7 @@ void FlatpakShim::OnProgressChanged(FlatpakTransactionProgress* progress,
   }
 }
 
-void FlatpakShim::OnNewOperation(FlatpakTransaction* transaction,
+void FlatpakShim::OnNewOperation(FlatpakTransaction* /* transaction */,
                                  FlatpakTransactionOperation* operation,
                                  FlatpakTransactionProgress* progress,
                                  gpointer user_data) {
@@ -3790,7 +3784,7 @@ void FlatpakShim::OnNewOperation(FlatpakTransaction* transaction,
   }
 }
 
-void FlatpakShim::OnOperationComplete(FlatpakTransaction* transaction,
+void FlatpakShim::OnOperationComplete(FlatpakTransaction* /* transaction */,
                                       FlatpakTransactionOperation* operation,
                                       const char* commit,
                                       FlatpakTransactionResult result,
@@ -3836,7 +3830,7 @@ void FlatpakShim::OnOperationComplete(FlatpakTransaction* transaction,
 }
 
 gboolean FlatpakShim::OnOperationError(
-    FlatpakTransaction* transaction,
+    FlatpakTransaction* /* transaction */,
     FlatpakTransactionOperation* operation,
     const GError* error,
     FlatpakTransactionErrorDetails error_details,

--- a/plugins/flatpak/flatpak_shim.h
+++ b/plugins/flatpak/flatpak_shim.h
@@ -50,6 +50,11 @@ struct FlatpakShim : std::enable_shared_from_this<FlatpakShim> {
                        asio::io_context::strand* strand = nullptr)
       : plugin_(plugin), messenger_(messenger), strand_(strand) {}
 
+  ~FlatpakShim() {
+    plugin_ = nullptr;
+    messenger_ = nullptr;
+  }
+
   /**
    * \brief Retrieves an optional attribute from an XML node.
    * \param node Pointer to the XML node.

--- a/plugins/flatpak/messages.g.cc
+++ b/plugins/flatpak/messages.g.cc
@@ -920,7 +920,7 @@ void FlatpakApi::SetUp(flutter::BinaryMessenger* binary_messenger,
         &GetCodec());
     if (api != nullptr) {
       channel.SetMessageHandler(
-          [api](const EncodableValue& message,
+          [api](const EncodableValue& /* message */,
                 const flutter::MessageReply<EncodableValue>& reply) {
             try {
               ErrorOr<EncodableList> output = api->GetApplicationsUpdate();

